src/nodes.coffee

```diff
@@ -1532,6 +1532,9 @@ exports.Param = class Param extends Base
         obj = obj.variable
       # * assignments within destructured parameters `{foo:bar}`
       if obj instanceof Assign
+        # ... possibly with a default value
+        if obj.value instanceof Assign
```

Comment:
@vendethiel It’s no more weird than `(a = b = 1) ->`, which has been around for ages.

I’ll add a test, then.